newIDE/app/src/CommandPalette/CommandPalette.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+import React from 'react';
+import { I18n } from '@lingui/react';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import Dialog from '@material-ui/core/Dialog';
+import TextField from '@material-ui/core/TextField';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import { makeStyles } from '@material-ui/core/styles';
+import CommandsContext from '../CommandPalette/CommandsContext';
+import { type NamedCommand } from '../CommandPalette/CommandManager';
+
+const useStyles = makeStyles({
+  scrollPaper: {
+    alignItems: 'flex-start',
+  },
+});
+
+type Props = {|
+  open: boolean,
+  onClose: () => void,
+|};
+
+const CommandPalette = (props: Props) => {
+  const [autocompleteOpen, openAutocomplete] = React.useState(true);
+  const classes = useStyles();
+  const { onClose, open } = props;
+  const commandManager = React.useContext(CommandsContext);
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  const handleCommandChoose = (e, command: NamedCommand) => {
+    command && command.handler();
+    props.onClose();
+  };
+
+  return (
+    <I18n>
+      {({ i18n }) => (
+        <Dialog
+          onClose={handleClose}
+          aria-label="command-palette"
+          open={open}
+          fullWidth
+          hideBackdrop
+          maxWidth="sm"
+          classes={classes}
+          transitionDuration={0}
+        >
+          <Autocomplete
+            open={autocompleteOpen}
+            onClose={() => {
+              openAutocomplete(false);
+              handleClose();
+            }}
+            onOpen={() => openAutocomplete(true)}
+            options={commandManager.getAllNamedCommands()}
+            getOptionLabel={command => i18n._(command.displayText)}
+            onChange={handleCommandChoose}
+            openOnFocus
+            renderInput={params => (
+              <TextField
+                {...params}
+                placeholder="Start typing a command..."
```

Comment:
We need to add translation marker to this :)
Is the placeholder a React node or a string?
If a React Node, use `<Trans>` component. Otherwise, call 
```
i18n._(t``)
```

to get a string to pass to the placeholder

Reply:
Right, forgot about that one. Done.